static/app/views/dashboardsV2/widgetCard/widgetCardChartContainer.tsx

```diff
@@ -0,0 +1,249 @@
+import * as React from 'react';
+import {withRouter, WithRouterProps} from 'react-router';
+import styled from '@emotion/styled';
+
+import {Client} from 'sentry/api';
+import ErrorPanel from 'sentry/components/charts/errorPanel';
+import SimpleTableChart from 'sentry/components/charts/simpleTableChart';
+import TransparentLoadingMask from 'sentry/components/charts/transparentLoadingMask';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import Placeholder from 'sentry/components/placeholder';
+import {IconWarning} from 'sentry/icons';
+import space from 'sentry/styles/space';
+import {Organization, PageFilters} from 'sentry/types';
+import {getIssueFieldRenderer} from 'sentry/utils/dashboards/issueFieldRenderers';
+import {TableDataRow} from 'sentry/utils/discover/discoverQuery';
+
+import {Widget, WidgetType} from '../types';
+import {
+  ISSUE_FIELD_TO_HEADER_MAP,
+  ISSUE_FIELDS,
+} from '../widgetBuilder/issueWidget/fields';
+
+import WidgetCardChart from './chart';
+import IssueWidgetQueries from './issueWidgetQueries';
+import MetricsWidgetQueries from './metricsWidgetQueries';
+import WidgetQueries from './widgetQueries';
+
+type TableResultProps = Pick<WidgetQueries['state'], 'errorMessage' | 'loading'> & {
+  transformedResults: TableDataRow[];
+};
+
+type Props = WithRouterProps & {
+  api: Client;
+  organization: Organization;
+  selection: PageFilters;
+  widget: Widget;
+  isMobile?: boolean;
+  renderErrorMessage?: (errorMessage?: string) => React.ReactNode;
+  tableItemLimit?: number;
+  windowWidth?: number;
+};
+
+class WidgetCardChartContainer extends React.Component<Props> {
```

Comment:
can we convert this class to a regular react function?

Reply:
👍  Updated to a functional component!